Deprecating sql_database access for creating UC functions for agent tools (#29745)
langchain-ai/langchain: docs/docs/integrations/providers/databricks.md

SQLDatabase
-----------

> You can connect to Databricks SQL or query structured data See [Structured retrieval AI agent tools](https://docs.databricks.com/en/generative-ai/agent-framework/structured-retrieval-tools.html#table-query-tool).

Reviewer: ```suggestion
To connect to Databricks SQL or query structured data, see the [Databricks structured retriever tool documentation](https://docs.databricks.com/en/generative-ai/agent-framework/structured-retrieval-tools.html#table-query-tool).
```

```

> from databricks_langchain.uc_ai import DatabricksFunctionClient

Reviewer: I might just leave this out, it'll be in the linked docs page (it'd be nice not to have to duplicate + keep content up to date across places)

SQLDatabase
-----------

> To connect to Databricks SQL or query structured data, see the [Databricks structured retriever tool documentation](https://docs.databricks.com/en/generative-ai/agent-framework/structured-retrieval-tools.html#table-query-tool).

Reviewer 2: Can we point folks to the relevant part(s) of `databricks-langchain`?

For example, does it make sense to link to the Databricks [integration page](https://python.langchain.com/docs/integrations/tools/databricks/), and update that page to also reference https://docs.databricks.com/en/generative-ai/agent-framework/structured-retrieval-tools.html ?

Author: ```suggestion
To connect to Databricks SQL or query structured data, see the [Databricks structured retriever tool documentation](https://docs.databricks.com/en/generative-ai/agent-framework/structured-retrieval-tools.html#table-query-tool) and to create an agent using the above created SQL UDF see [Databricks UC Integration](https://python.langchain.com/docs/integrations/tools/databricks/)
```

Author: Thoughts on the suggestion?

> To connect to Databricks SQL or query structured data, see the [Databricks structured retriever tool documentation](https://docs.databricks.com/en/generative-ai/agent-framework/structured-retrieval-tools.html#table-query-tool) and to create an agent using the above created SQL UDF see [Databricks UC Integration](https://python.langchain.com/docs/integrations/tools/databricks/).

Author: ```suggestion
To connect to Databricks SQL or query structured data, see the [Databricks structured retriever tool documentation](https://docs.databricks.com/en/generative-ai/agent-framework/structured-retrieval-tools.html#table-query-tool) and to create an agent using the above created SQL UDF see [Databricks UC Integration](https://docs.unitycatalog.io/ai/integrations/langchain/).
```
